src/compile/layer.ts

```diff
@@ -1,36 +1,69 @@
-import {Channel} from '../channel';
-import {keys, duplicate, mergeDeep, flatten, unique, isArray, vals, hash, Dict} from '../util';
+import {Channel, CHANNELS} from '../channel';
+import {keys, flatten, unique, Dict, forEach, extend} from '../util';
 import {defaultConfig, Config} from '../config';
-import {LayerSpec} from '../spec';
+import {LayerSpec, ResolveMapping, Resolve, INDEPENDENT, SHARED} from '../spec';
 import {assembleData, parseLayerData} from './data/data';
 import {assembleLayout, parseLayerLayout} from './layout';
 import {Model} from './model';
 import {UnitModel} from './unit';
 import {buildModel} from './common';
 import {FieldDef} from '../fielddef';
-import {ScaleComponents} from './scale';
-import {VgData, VgAxis, VgLegend, isUnionedDomain, isDataRefDomain, VgDataRef} from '../vega.schema';
+import {ScaleComponents, mergeScales} from './scale';
+import {VgData, VgAxis, VgLegend} from '../vega.schema';
 
 
 export class LayerModel extends Model {
   private _children: UnitModel[];
 
+  // resolve for every channel used in children
+  private _resolve: ResolveMapping;
+
   constructor(spec: LayerSpec, parent: Model, parentGivenName: string) {
     super(spec, parent, parentGivenName);
 
-    this._config = this._initConfig(spec.config, parent);
     this._children = spec.layers.map((layer, i) => {
       // we know that the model has to be a unit model beacuse we pass in a unit spec
       return buildModel(layer, this, this.name('layer_' + i)) as UnitModel;
     });
+
+    this._resolve = this._initResolve(spec.resolve || {});
   }
 
-  private _initConfig(specConfig: Config, parent: Model) {
-    return mergeDeep(duplicate(defaultConfig), specConfig, parent ? parent.config() : {});
+  /**
+   * Figure out which fields have different field types and make those independent.
+   * Use resolve for all other fields with default shared.
+   */
+  private _initResolve(resolve: ResolveMapping): ResolveMapping {
+
+    for (var i = 0; i < CHANNELS.length; i++) {
+      var channel = CHANNELS[i];
+      var types = unique(this.children().map((child: UnitModel) => {
+        const fd = child.fieldDef(channel);
+        if (fd) {
+          return fd.type;
+        }
+        return undefined;
+      }).filter((type) => !!type));
+
+      if (types.length > 1) {
+        resolve[channel] = {
+          scale: INDEPENDENT,
+          guide: INDEPENDENT
```

Comment:
We don't use the term `guide` anywhere else in the syntax. 

I think we should use `axis` and `legend` here.   (depending on type of channel)